docs/design/2018-10-24-view-support.md

```diff
@@ -5,65 +5,128 @@
 
 Below is a visual depiction of a view:
-  ![AnatomyOfAview](imgs/view.png)
+  ![AnatomyOfAview](imgs/view.png)   
+  reference from https://www.essentialsql.com/what-is-a-relational-database-view/
   
 A view is created from a query using the "`CREATE OR REPLACE VIEW`" command. In the example below we are creating a PopularBooks view based on a query which selects all Books that have the IsPopular field checked. Following is the query:
-`CREATE OR REPLACE VIEW PopularBooks AS SELECT ISBN, Title, Author,PublishDate FROM Books WHERE IsPopular=1`  
+```mysql
+CREATE OR REPLACE 
+  VIEW PopularBooks AS 
+    SELECT ISBN, Title, Author,PublishDate FROM Books WHERE IsPopular=1
+```  
 
 Once a view is created, you can use it as you query any table in a `SELECT` statement. For example, to list all the popular book titles ordered by an author, you could write:  
-`SELECT Author, Title FROM PopularBooks ORDER BY Author`  
+```mysql
+SELECT Author, Title FROM PopularBooks ORDER BY Author
+```
 
 In general you can use any of the SELECT clauses, such as GROUP BY, in a select statement containing a view.
 
 ## Proposal
-I implement the "`CREATE OR REPLACE`" command to create a view with the `SELECT` statement. Currently, I reuse "`DROP TABLE`" to drop a view, and "`SHOW FULL TABLES`" can also display the view list with different table types. The "`SHOW CREATE TABlE`" command can also regenerate a view create command.
+This proposal is prepared to implement basic VIEW feature, which contains "CREATE OR REPLACE VIEW", "SELECT FROM VIEW", "DROP VIEW" and "SHOW TABLE STATUS". All other unimplemented feature will list as compatibility and discuss later.
+and we mainly introduce new struct named `ViewInfo` to store view's metadata.
 
 ## Rationale
-`VIEW` is just a tableinfo object store with a `SELECT` statement, and only supports two types of DDL operations, which are "`CREATE OR REPLACE VIEW`" and "`DROP`". Currently, view can only be used within the `SELECT` statement.  
+`VIEW` is just a `TableInfo` object store with a `ViewInfo` struct, and only supports limited DDL operations.  
+Here is `ViewInfo`'s attributes and detail attribute explaination:
+```
+type ViewInfo struct {
+	Algorithm   AlgorithmType    `json:"view_algorithm"`
+	Definer     string           `json:"view_definer"`  
+	Security    SecurityType     `json:"view_security""`
+	SelectStmt  string           `json:"view_select"`
+	CheckOption CheckOption      `json:"view_checkoption"`
+	Cols        []string         `json:"view_cols"`
+}
+```
+* AlgorithmType  ref https://dev.mysql.com/doc/refman/5.7/en/view-algorithms.html   
+    The view SQL AlGORITHM characteristic. The value is one of UNDEFINED、MERGE OR TEMPTABLE, if no ALGORITHM clause is present, UNDEFINED is the default algorithm.
+    We will implement Algorithm=UNDEFINED only now.
```

Comment:
We will implement MERGE not UNDEFINED, and regard UNDEFINED as MERGE now.
As the MySQL manual shows:
```
If no ALGORITHM clause is present, UNDEFINED is the default algorithm prior to MySQL 5.7.6. As of 5.7.6, the default algorithm is determined by the value of the derived_merge flag of the optimizer_switch system variable. For additional discussion, see Section 8.2.2.3, “Optimizing Derived Tables and View References”.
```